pkg/controller/aws/network/internetgateway.go

```diff
@@ -0,0 +1,234 @@
+	"github.com/crossplaneio/crossplane/pkg/controller/aws/utils"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+)
+
+const (
+	errObjectNotInternetGateway = "The managed resource is not an InternetGateway resource"
+)
+
+// InternetGatewayController is the controller for InternetGateway objects
+type InternetGatewayController struct{}
+
+// SetupWithManager creates a new Controller and adds it to the Manager with default RBAC. The Manager will set fields on the Controller
+// and Start it when the Manager is Started.
+func (c *InternetGatewayController) SetupWithManager(mgr ctrl.Manager) error {
+	r := resource.NewManagedReconciler(mgr,
+		resource.ManagedKind(v1alpha1.InternetGatewayGroupVersionKind),
+		resource.WithExternalConnecter(&igConnector{client: mgr.GetClient()}),
+		resource.WithManagedConnectionPublishers())
+
+	name := strings.ToLower(fmt.Sprintf("%s.%s", v1alpha1.InternetGatewayKindAPIVersion, v1alpha1.Group))
+
+	return ctrl.NewControllerManagedBy(mgr).
+		Named(name).
+		For(&v1alpha1.InternetGateway{}).
+		Complete(r)
+}
+
+type igConnector struct {
+	client      client.Client
+	newClientFn func(*aws.Config) (ec2.InternetGatewayClient, error)
+}
+
+func (conn *igConnector) Connect(ctx context.Context, mgd resource.Managed) (resource.ExternalClient, error) {
+	cr, ok := mgd.(*v1alpha1.InternetGateway)
+	if !ok {
+		return nil, errors.New(errObjectNotInternetGateway)
+	}
+
+	newClientFn := ec2.NewInternetGatewayClient
+	if conn.newClientFn != nil {
+		newClientFn = conn.newClientFn
+	}
+
+	awsconfig, err := utils.RetrieveAwsConfigFromProvider(ctx, conn.client, cr.Spec.ProviderReference)
+	if err != nil {
+		return nil, err
+	}
+
+	c, err := newClientFn(awsconfig)
+	if err != nil {
+		return nil, errors.Wrap(err, "cannot create a new InternetGatewayClient")
+	}
+
+	return &igExternal{c}, nil
+}
+
+type igExternal struct {
+	client ec2.InternetGatewayClient
+}
+
+func (e *igExternal) Observe(ctx context.Context, mgd resource.Managed) (resource.ExternalObservation, error) {
+	cr, ok := mgd.(*v1alpha1.InternetGateway)
+	if !ok {
+		return resource.ExternalObservation{}, errors.New(errObjectNotInternetGateway)
+	}
+
+	// To find out whether an InternetGateway exist:
+	// - the object's ExternalState should have internetGatewayID populated
+	// - an InternetGateway with the given internetGatewayID should exist
+	if cr.Status.InternetGatewayID == "" {
+		return resource.ExternalObservation{
+			ResourceExists: false,
+		}, nil
+	}
+
+	req := e.client.DescribeInternetGatewaysRequest(&awsec2.DescribeInternetGatewaysInput{
+		InternetGatewayIds: []string{cr.Status.InternetGatewayID},
+	})
+	req.SetContext(ctx)
+
+	response, err := req.Send()
+
+	if err != nil {
+		if ec2.IsInternetGatewayNotFoundErr(err) {
+			return resource.ExternalObservation{
+				ResourceExists: false,
+			}, nil
+		}
+
+		return resource.ExternalObservation{}, errors.Wrapf(err, "Error describing InternetGateway with id: %v", cr.Status.InternetGatewayID)
```

Comment:
https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Error strings should not be capitalized; I thought this was something our linter caught.